src/com/owncloud/android/datamodel/ThumbnailsCacheManager.java

```diff
@@ -359,7 +358,223 @@ private Bitmap doFileInBackground() {
+                        if (mUsername instanceof String) {
+                            tagId = (String) mUsername;
+                            if (String.valueOf(imageView.getTag()).equals(tagId)) {
+                                imageView.setImageBitmap(bitmap);
+                            }
+                        }
+                    }
+                } else {
+                    MenuItem menuItem = mMenuItemReference.get();
+                    AvatarGenerationTask avatarWorkerTask = getAvatarWorkerTask(menuItem);
+                    if (this == avatarWorkerTask) {
+                        String tagId = "";
+                        if (mUsername instanceof String) {
+                            tagId = (String) mUsername;
+                            if (String.valueOf(menuItem.getTitle()).equals(tagId)) {
+                                menuItem.setIcon(new BitmapDrawable(MainApp.getAppContext().getResources(),
+                                        bitmap));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        /**
+         * Add thumbnail to cache
+         * @param imageKey: thumb key
+         * @param bitmap:   image for extracting thumbnail
+         * @param path:     image path
+         * @param px:       thumbnail dp
+         * @return Bitmap
+         */
+        private Bitmap addThumbnailToCache(String imageKey, Bitmap bitmap, String path, int px){
+
+            Bitmap thumbnail = ThumbnailUtils.extractThumbnail(bitmap, px, px);
+
+            // Rotate image, obeying exif tag
+            thumbnail = BitmapUtils.rotateImage(thumbnail,path);
+
+            // Add thumbnail to cache
+            addBitmapToCache(imageKey, thumbnail);
+
+            return thumbnail;
+        }
+
+        /**
+         * Converts size of file icon from dp to pixel
+         * @return int
+         */
+        private int getAvatarDimension(){
+            // Converts dp to pixel
+            Resources r = MainApp.getAppContext().getResources();
+            return Math.round(r.getDimension(R.dimen.file_avatar_size));
+        }
+
+        private Bitmap doAvatarInBackground() {
+            String username = (String) mUsername;
+
+            final String imageKey = "a_" + username;
+
+            // Check disk cache in background thread
+            Bitmap avatar = getBitmapFromDiskCache(imageKey);
+
+            // Not found in disk cache
+            if (avatar == null) {
+
+                int px = getAvatarDimension();
+
+                // Download avatar from server
+                OwnCloudVersion serverOCVersion = AccountUtils.getServerVersion(mAccount);
+                if (mClient != null && serverOCVersion != null) {
+                    if (serverOCVersion.supportsRemoteThumbnails()) {
+                        GetMethod get = null;
+                        try {
+                            String uri = mClient.getBaseUri() + "" +
```

Comment:
Tricky... there is no official Avatar API... The implementation simply "knows" where to look for the avatar image and fetches it.